### PR TITLE
Add support for generating `robolectric_rule` for Android library tests.

### PR DESCRIPTION
**Pull request by raviagarwal7**
- Added `source_under_test` & `tests` property support
 - example of a generated rule.
```
robolectric_test(
	name = 'test_release',
	srcs = glob([
		'src/test/java/**/*.java',
	]),
	source_under_test = [
		':src_release',
	],
	manifest = 'build/okbuck/release/AndroidManifest.xml',
	annotation_processors = [
		'dagger.internal.codegen.ComponentProcessor',
		...
	],
	annotation_processor_deps = [
		':apt_jar_release',
	],
	provided_deps = [
		'//.okbuck/cache/1ad14065391dbef088f8bb7624b66135:dagger-compiler-2.5.jar',
		...
	],
	source = '7',
	target = '7',
	runtime_dependency = '.okbuck/cache/robolectric',
	deps = [
		':src_release',
		'//.okbuck/cache/b617945ee6b41679ad510a16e3851a36:design-24.0.0.aar',
		'//.okbuck/cache/18dafd39d53912ce093108cc6a5eb4f0:appcompat-v7-24.0.0.aar',
		...
	],
	visibility = [
		'PUBLIC',
	],
)
```

### CODE REVIEW

File: buildSrc/src/main/groovy/com/github/okbuilds/core/model/AndroidTarget.groovy [hunk 1]
```diff
@@ -65,12 +66,12 @@ abstract class AndroidTarget extends JavaLibTarget {
...
-                ["testCompile", "${buildType}TestCompile", "${flavor}TestCompile", "${name}TestCompile"] as Set,
-                baseVariant.sourceSets.collect { SourceProvider provider ->
-                    provider.javaDirectories
-                }.flatten() as Set<File>,
-                null,
+        return new Scope(
+                project,
+                ["compile", "${buildType}Compile", "${flavor}Compile", "${name}Compile",
+                 "testCompile", "${buildType}TestCompile", "${flavor}TestCompile", "${name}TestCompile"] as Set,
+                project.files("src/test/java") as Set<File>,
```
Comment (kageiit) [inline buildSrc/src/main/groovy/com/github/okbuilds/core/model/AndroidTarget.groovy]: how about using the basevariant api to get the test sourcedirs and resourcedirs?

**Comment by kageiit** [comment]
Can we fix the glob handler for test sources to be

`glob(['**/*Test.java'])` instead of `glob(['**/*.java'])`

File: buildSrc/src/main/groovy/com/github/okbuilds/core/model/AndroidTarget.groovy (hunk 1 above)
Comment (raviagarwal7) [inline buildSrc/src/main/groovy/com/github/okbuilds/core/model/AndroidTarget.groovy]: discussed offline, testVariant is null for library variants so we will keep as is.

File: buildSrc/src/main/groovy/com/github/okbuilds/core/model/AndroidTarget.groovy [hunk 2]
```diff
@@ -75,6 +76,13 @@ abstract class AndroidTarget extends JavaLibTarget {
                 extraJvmArgs)
     }
 
+    public boolean getRobolectric() {
+        ExperimentalExtension experimental = rootProject.okbuck.experimental
+        if (experimental.robolectric) {
```
Comment (kageiit) [inline buildSrc/src/main/groovy/com/github/okbuilds/core/model/AndroidTarget.groovy]: simplify to 
`return experimental.robolectric`

**Pull request merged**